clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -870,10 +875,14 @@ void CIRGenModule::emitGlobalVarDefinition(const clang::VarDecl *vd,
     emitter->finalize(gv);
 
   // If it is safe to mark the global 'constant', do so now.
+  // Use the same logic as emitCXXGlobalVarDeclInit to determine constant
```

Comment:
This code does not use the same logic as emitCXXGlobalVarDeclInit in classic codegen. I don't know why it doesn't but it doesn't. This change could lead us to behave differently.